### PR TITLE
MNT: Create and test universal wheels

### PR DESCRIPTION
**Pull request by effigies**
This removes the cibuildwheel infrastructure needed to compile C extensions. Instead, a single wheel is built and then explicitly tested on Python 2 and 3 across the three operating systems supported by GitHub Actions.

### CODE REVIEW

**Comment by coalsont** [comment]
Should this also wait on PyPI releasing the package name?

**Comment by effigies** [comment]
No, this can go in as-is. But it would be good to wait on pypi to release.

**Comment by bpinsard** [comment]
Cannot see gradunwarp on pypi, so I guess the name has been released?

**Comment by effigies** [comment]
Nope. Wonder if we should push `gradunwarp-hcp` package until then?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -68,3 +68,86 @@ jobs:
...
+      - name: Set up Python ${{ matrix.python[1] }} (native)
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python[1] }}
+          allow-prereleases: true
+        if: matrix.python[0] == 'native'
+      - name: Set up Python ${{ matrix.python[1] }} (miniconda)
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          python-version: ${{ matrix.python[1] }}
```
Comment (effigies) [inline .github/workflows/test.yml:122]:
```suggestion
          miniforge-version: latest
          python-version: ${{ matrix.python[1] }}
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -68,3 +68,87 @@ jobs:
...
+      - name: Set up Python ${{ matrix.python[1] }} (native)
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python[1] }}
+          allow-prereleases: true
+        if: matrix.python[0] == 'native'
+      - name: Set up Python ${{ matrix.python[1] }} (miniconda)
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-version: latest
```
Comment (effigies) [inline .github/workflows/test.yml]:
```suggestion
          # OSX runners do not have a built-in conda, so install, but don't change other OSs from default
          miniforge-version: ${{ startsWith(matrix.os, "macos") && "latest" || "" }}
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -68,3 +68,88 @@ jobs:
...
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python[1] }}
+          allow-prereleases: true
+        if: matrix.python[0] == 'native'
+      - name: Set up Python ${{ matrix.python[1] }} (miniconda)
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          # OSX runners do not have a built-in conda, so install, but don't change other OSs from default
+          miniforge-version: ${{ startsWith(matrix.os, "macos") && "latest" || "" }}
```
Comment (effigies) [inline .github/workflows/test.yml]:
```suggestion
          miniforge-version: ${{ startsWith(matrix.os, 'macos') && 'latest' || '' }}
```

**Pull request merged**